lisa/base_tools/uname.py

```diff
@@ -8,10 +8,34 @@
 from lisa.util import LisaException
 
 
+@dataclass
+class KernelVersion:
+    major: int = 0
+    minor: int = 0
+    patch: int = 0
+    kernel_version: str = ""
+
+    def __init__(self, kernel_version: str) -> None:
+        if not kernel_version:
+            return
+        self.kernel_version = kernel_version
+        version_string = kernel_version.split("-")[0]
+        version = [int(num) for num in version_string.split(".")]
+        self.major = version[0]
+        self.minor = version[1]
+        self.patch = version[2]
+
+    def __repr__(self) -> str:
+        return self.kernel_version
+
+    def __str__(self) -> str:
+        return self.kernel_version
+
+
 @dataclass
 class UnameResult:
     has_result: bool
-    kernel_version: str = ""
+    kernel_version: KernelVersion
```

Comment:
Let's keep both parsed structured version and original string.

Reply:
`KernelVersion` class has original string and I have overridden `__str__` and `__repr__` methods that return original string. This ensures that the code works without making any change elsewhere. Any reason to keep them separate?

Reply:
The raw string may be logged. And the structure is not able to dump the raw string, if we use the class of semver.